### PR TITLE
upgrade argo-cd version to 2.12.1

### PR DESCRIPTION
**Pull request by ishitasequeira**
**What type of PR is this?**
/kind chore

**What does this PR do / why we need it**:

upgrade argo-cd version to 2.12.1

### CODE REVIEW

File: bundle/manifests/argoproj.io_argocdexports.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.6.1
```
Comment (anandf) [inline bundle/manifests/argoproj.io_argocdexports.yaml:5]: Can you run `make bundle` command with the correct version of `controller-gen` - `0.14.0`

**Review by anandf — CHANGES_REQUESTED** [review]
Update controller-gen version to 0.14.0

**Review by saumeya — APPROVED** [review]
LGTM

**Review by svghadi — CHANGES_REQUESTED** [review]
Dockerfile also needs an update, rest all looks good. Thanks.
https://github.com/argoproj-labs/argocd-operator/blob/6a5b0817a758bdc531369f7eb7c20a118640762a/build/util/Dockerfile#L1-L2

**Review by anandf — COMMENTED** [review]
Are the YAML files in https://github.com/argoproj-labs/argocd-operator/tree/master/bundle/manifests updated to take the upstream ArgoCD v2.12.1 ?

**Comment by svghadi** [comment]
Oh, yes. We also need to update the Argo CD Application, Project & ApplicationSet CRD in https://github.com/argoproj-labs/argocd-operator/tree/master/config/crd/bases. Just copy these files from upstream Argo CD source tag for new version.

**Review by anandf — COMMENTED** [review]
LGTM

**Review by anandf — APPROVED** [review]

**Comment by svghadi** [comment]
> Because when I replace the CRDs with 2.12.1 and execute make bundle, the CRDs revert back to it's current state. 

Interesting. I see some latest changes in Application and Appset CRD in upstream https://github.com/argoproj/argo-cd/tree/v2.12.1/manifests/crds which are not present in our copy of CRDs here: https://github.com/argoproj-labs/argocd-operator/tree/master/config/crd/bases .

**Review by svghadi — APPROVED** [review]
LGTM.. Thanks

File: go.mod [hunk 2]
```diff
@@ -1,9 +1,11 @@
 module github.com/argoproj-labs/argocd-operator
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.9
```
Comment (anandf) [inline go.mod:5]: I think we can remove this.

Comment (svghadi) [reply go.mod:5]: Not sure how to do it. `go mod tidy` is adding it again even if I remove it and set to just `go 1.21`.

**Comment by Invincibear** [comment]
Can we get this pushed to https://operatorhub.io/operator/argocd-operator?

**Comment by Invincibear** [comment]
Is this ready for a new release so we can get this pushed to https://operatorhub.io/operator/argocd-operator?

**Comment by svghadi** [comment]
Hi @Invincibear, it will still take a few weeks to get a release out. In the meantime, you can use the `.spec.image` and `.spec.version` fields in the `ArgoCD` CR to specify the new image to unblock yourself. Since you mentioned that there are some bug fixes in the new images that you're interested in, you likely won't run into any issues using the new image with the old operator, as the fixes, most of the times, don't require new Argo CD CRDs.

**Comment by Invincibear** [comment]
@svghadi Thanks, that worked!

**Pull request merged**